azurerm/internal/services/authorization/parse/role_assignment_test.go

```diff
@@ -176,6 +196,18 @@ func TestRoleAssignmentID(t *testing.T) {
 				TenantId:        "34567812-3456-7653-6742-345678901234",
 			},
 		},
+		{
+			Input: "/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.Storage/storageAccounts/nameStorageAccount/providers/Microsoft.Authorization/roleAssignments/23456781-2349-8764-5631-234567890121|34567812-3456-7653-6742-345678901234",
+			Expected: &RoleAssignmentId{
+				SubscriptionID:   "12345678-1234-9876-4563-123456789012",
+				ResourceGroup:    "group1",
+				ResourceProvider: "Microsoft.Storage",
+				ResourceScope:    "storageAccounts/nameStorageAccount",
+				ManagementGroup:  "",
+				Name:             "23456781-2349-8764-5631-234567890121",
+				TenantId:         "34567812-3456-7653-6742-345678901234",
+			},
+		},
```

Comment:
we should check whether the `ResourceProvider` and `ResourceScope` equals to the expected value

Reply:
Thanks, missed that indeed! 😄